fg_lib/doc/api.rst

```diff
@@ -0,0 +1,17 @@
+.. _fuel_gauge_api:
+
+API documentation
+#################
+
+.. contents::
+   :local:
+   :depth: 2
+
+Fuel Gauge API documentation
+****************************
+
+| Header file: :file:`include/fg_lib.h`
+
+.. doxygengroup:: Fuel Gauge
```

Comment:
```suggestion
.. doxygengroup:: fg_lib
```
You need the tag from the @defgroup in the header here. It will pull the name of the group: "Battery fuel gauge library" together with all the API.